File: schemaorg/musicalbumproductiontype.go
package schemaorg

import "encoding/json"

// MusicAlbumProductionType see : https://schema.org/MusicAlbumProductionType
type MusicAlbumProductionType struct {

Enumeration

typeContext

// SupersededBy see : http://meta.google.schema.org/supersededBy
// Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
SupersededBy interface{} `json:"supersededBy"` // types : Class Enumeration Property

}

func (v *MusicAlbumProductionType) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "MusicAlbumProductionType"

	return json.Marshal(v)
}

File: schemaorgphp/classes/music_album_production_type.php
<?php

class MusicAlbumProductionType extends Enumeration implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'MusicAlbumProductionType';
	
	/**
	 * Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	 * see : https://meta.schema.org/supersededBy
	 * @var \Class|\Class[]|\Enumeration|\Enumeration[]|\Property|\Property[]
	 */
	public var $superseded_by;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'MusicAlbumProductionType'
		);
		
		$serialized = so_json_serialize( $this->superseded_by );
		if ( ! empty( $serialized ) ) {
			$out['supersededBy'] = $serialized;
		}
		
		return $out;
	}
}
